src/tools/lifter/bir_lifting_machinesScript.sml

```diff
@@ -746,4 +747,147 @@ val bmr_ms_mem_contains_M0_2 = store_thm ("bmr_ms_mem_contains_M0_2",
+    (BMLI (BVar "R7" (BType_Imm Bit32)) (\ms:m0_mod_state. Imm32 (ms.base.REG (R_name T 7w))));
+    (BMLI (BVar "R8" (BType_Imm Bit32)) (\ms:m0_mod_state. Imm32 (ms.base.REG (R_name T 8w))));
+    (BMLI (BVar "R9" (BType_Imm Bit32)) (\ms:m0_mod_state. Imm32 (ms.base.REG (R_name T 9w))));
+    (BMLI (BVar "R10" (BType_Imm Bit32)) (\ms:m0_mod_state. Imm32 (ms.base.REG (R_name T 10w))));
+    (BMLI (BVar "R11" (BType_Imm Bit32)) (\ms:m0_mod_state. Imm32 (ms.base.REG (R_name T 11w))));
+    (BMLI (BVar "R12" (BType_Imm Bit32)) (\ms:m0_mod_state. Imm32 (ms.base.REG (R_name T 12w))));
+    (BMLI (BVar "LR" (BType_Imm Bit32)) (\ms:m0_mod_state. Imm32 (ms.base.REG (R_name T 14w))));
+    (BMLI (BVar "SP_main" (BType_Imm Bit32)) (\ms:m0_mod_state. Imm32 (ms.base.REG (R_name F 13w))));
+    (BMLI (BVar "SP_process" (BType_Imm Bit32)) (\ms:m0_mod_state. Imm32 (ms.base.REG (R_name T 13w))));
+    (BMLI (BVar "ModeHandler" BType_Bool) (\ms:m0_mod_state. bool2b (ms.base.CurrentMode = Mode_Handler)));
+    (BMLI (BVar "countw" (BType_Imm Bit64)) (\ms:m0_mod_state. Imm64 (ms.countw)))]`;
+
+
+val m0_mod_REGS_lifted_imms_LIST_REWRS = save_thm ("m0_mod_REGS_lifted_imms_LIST_REWRS",
+  SIMP_RULE (std_ss++wordsLib.WORD_ss) [R_name_def] m0_mod_REGS_lifted_imms_LIST_def);
+
+val m0_mod_lifted_mem_def = Define `
+  m0_mod_lifted_mem = BMLM (BVar "MEM" (BType_Mem Bit32 Bit8)) (\ms:m0_mod_state. ms.base.MEM)`
+
+val m0_mod_lifted_pc_def = Define `
+  m0_mod_lifted_pc = BMLPC (BVar (bir_temp_var_name "PC") (BType_Imm Bit32))
+                       (BVar (bir_temp_var_name "COND") BType_Bool)
+                       (\ms:m0_mod_state. Imm32 (ms.base.REG RName_PC))`
+
+val m0_mod_state_is_OK_def = Define `m0_mod_state_is_OK (ef, sel) (s:m0_mod_state) =
+  ((s.base.AIRCR.ENDIANNESS <=> ef) /\ (s.base.CONTROL.SPSEL <=> sel) /\
+  (s.base.exception = NoException))`
+
+(* Just a dummy for now *)
+val m0_mod_bmr_def = Define `m0_mod_bmr (ef, sel) = <|
+  bmr_extra := \ms:m0_mod_state. m0_mod_state_is_OK (ef, sel) ms;
+  bmr_imms := m0_mod_REGS_lifted_imms_LIST;
+  bmr_mem := m0_mod_lifted_mem;
+  bmr_pc := m0_mod_lifted_pc;
+  bmr_step_fun := NextStateM0_mod |>`;
+
+
+val m0_mod_bmr_EVAL = save_thm ("m0_mod_bmr_EVAL",
+  GENL [``ef:bool``, ``sel:bool``] (SIMP_CONV (list_ss++wordsLib.WORD_ss) [m0_mod_bmr_def, R_name_def,
+    m0_mod_REGS_lifted_imms_LIST_def, m0_mod_lifted_mem_def, m0_mod_state_is_OK_def,
+    m0_mod_lifted_pc_def, bir_temp_var_name_def]
+    ``m0_mod_bmr (ef, sel)``)
+);
+
+val m0_mod_bmr_vars_EVAL = save_thm ("m0_mod_bmr_vars_EVAL",
+SIMP_CONV (list_ss++bmr_ss) [m0_mod_bmr_EVAL, bmr_vars_def] ``bmr_vars (m0_mod_bmr (ef, sel))``);
+
+val m0_mod_bmr_temp_vars_EVAL = save_thm ("m0_mod_bmr_temp_vars_EVAL",
+SIMP_CONV (list_ss++bmr_ss) [m0_mod_bmr_EVAL, bmr_vars_def, bmr_temp_vars_def,
+  bir_temp_var_def, bir_temp_var_name_def]
+  ``bmr_temp_vars (m0_mod_bmr (ef, sel))``);
+
+val m0_mod_bmr_varnames_distinct = prove (``
```

Comment:
Those ` `` ` should go to the beginning of the following line.

Reply:
yes